crates/bevy_animation/src/animation_curves.rs

```diff
@@ -0,0 +1,544 @@
+        let new_value = self.0.sample_clamped(t);
+        component.scale = <Vec3 as Animatable>::interpolate(&component.scale, &new_value, weight);
+        Ok(())
+    }
+}
+
+/// This type allows a [curve] valued in `Transform` to become an [`AnimationCurve`] that animates
+/// a transform.
+///
+/// This exists primarily as a convenience to animate entities using the entire transform at once
+/// instead of splitting it into pieces and animating each part (translation, rotation, scale).
+///
+/// [curve]: Curve
+#[derive(Debug, Clone, Reflect, FromReflect)]
+#[reflect(from_reflect = false)]
+pub struct TransformCurve<C>(pub C);
+
+impl<C> AnimationCurve for TransformCurve<C>
+where
+    C: InnerAnimationCurve<Transform>,
+{
+    fn clone_value(&self) -> Box<dyn AnimationCurve> {
+        Box::new(self.clone())
+    }
+
+    fn domain(&self) -> Interval {
+        self.0.domain()
+    }
+
+    fn apply<'a>(
+        &self,
+        t: f32,
+        transform: Option<Mut<'a, Transform>>,
+        _entity: EntityMutExcept<'a, (Transform, AnimationPlayer, Handle<AnimationGraph>)>,
+        weight: f32,
+    ) -> Result<(), AnimationEvaluationError> {
+        let mut component = transform.ok_or_else(|| {
+            AnimationEvaluationError::ComponentNotPresent(TypeId::of::<Transform>())
+        })?;
+        let new_value = self.0.sample_clamped(t);
+        *component = <Transform as Animatable>::interpolate(&component, &new_value, weight);
+        Ok(())
+    }
+}
+
+/// This type allows an [`IterableCurve`] valued in `f32` to be used as an [`AnimationCurve`]
+/// that animates [morph weights].
+///
+/// [morph weights]: MorphWeights
+#[derive(Debug, Clone, Reflect, FromReflect)]
+#[reflect(from_reflect = false)]
+pub struct WeightsCurve<C>(pub C);
+
+impl<C> AnimationCurve for WeightsCurve<C>
+where
+    C: IterableCurve<f32> + Debug + Clone + Reflectable,
+{
+    fn clone_value(&self) -> Box<dyn AnimationCurve> {
+        Box::new(self.clone())
+    }
+
+    fn domain(&self) -> Interval {
+        self.0.domain()
+    }
+
+    fn apply<'a>(
+        &self,
+        t: f32,
+        _transform: Option<Mut<'a, Transform>>,
+        mut entity: EntityMutExcept<'a, (Transform, AnimationPlayer, Handle<AnimationGraph>)>,
+        weight: f32,
+    ) -> Result<(), AnimationEvaluationError> {
+        let mut dest = entity.get_mut::<MorphWeights>().ok_or_else(|| {
+            AnimationEvaluationError::ComponentNotPresent(TypeId::of::<MorphWeights>())
+        })?;
+        lerp_morph_weights(dest.weights_mut(), self.0.sample_iter_clamped(t), weight);
+        Ok(())
+    }
+}
+
+/// Update `morph_weights` based on weights in `incoming_weights` with a linear interpolation
+/// on `lerp_weight`.
+fn lerp_morph_weights(
+    morph_weights: &mut [f32],
+    incoming_weights: impl Iterator<Item = f32>,
+    lerp_weight: f32,
+) {
+    let zipped = morph_weights.iter_mut().zip(incoming_weights);
+    for (morph_weight, incoming_weights) in zipped {
```

Comment:
We should have a TODO or an issue filed to SIMDify this at some point.

Reply:
This is a good idea. I'll submit an issue for it when this is merged.